Source/Core/loadArrayBuffer.js

```diff
@@ -28,13 +34,26 @@ define([
      * @see {@link http://www.w3.org/TR/cors/|Cross-Origin Resource Sharing}
      * @see {@link http://wiki.commonjs.org/wiki/Promises/A|CommonJS Promises/A}
      */
-    function loadArrayBuffer(url, headers, request) {
-        return loadWithXhr({
-            url : url,
-            responseType : 'arraybuffer',
-            headers : headers,
-            request : request
+    function loadArrayBuffer(urlOrResource, headers, request) {
+        //>>includeStart('debug', pragmas.debug);
+        Check.defined('urlOrResource', urlOrResource);
+        //>>includeEnd('debug');
+
+        if (defined(headers)) {
+            deprecationWarning('loadArrayBuffer.headers', 'The headers parameter has been deprecated. Set the headers property on the Resource parameter.');
+        }
+
+        if (defined(request)) {
+            deprecationWarning('loadArrayBuffer.request', 'The request parameter has been deprecated. Set the request property on the Resource parameter.');
+        }
+
+        var resource = Resource.createIfNeeded(urlOrResource, {
+            headers: headers,
+            request: request
         });
+
+        resource.responseType = 'arraybuffer';
```

Comment:
Why is this set here instead of passed into `createIfNeeded`?

Reply:
Nevermind, I see we want to set it in all cases; but it's a little weird that we modify the object that the user passed in and goes back to my concern over multiple calls using the same Resource instance.  Should we create a dervied resource to set the `responseType`?  CC @shunter

Reply:
Another possible option would be to keep `reponseType` as a separate parameter to `loadWithXhr`.

Reply:
Now that we clone them, this isn't an issue.